feat: Json parsing 관련 유틸리티 클래스 기능개발 및 테스트코드 작성

jojojojocho opened the pull request:
# feat: Json parsing 관련 유틸리티 클래스 기능개발 및 테스트코드 작성

## 개발내용

- Json 파싱을 위해서 JsonUtil 클래스 생성
- toJson, fromJson 메서드 개발 toJson 메서드: Object를 Json String으로 변환
- fromJson(String) 메서드: Json String을 Object로 변환
- fromJson(Reader) 메서드: Reader를 받아서 라인별로 문자열을 읽어 Object로 반환
- Gson을 이용하여 직렬화, 역직렬화 시행시 String 형식의 날짜 데이터를 LocalDate형식으로 변환하기 위한 inner 클래스(LocalDateTimeAdapter, LocalDateAdapter, LocalTimeAdapter) 생성
- toJson, fromJson에 관련한 테스트 코드 작성

jojojojocho committed:
feat: Json parsing 관련 유틸리티 클래스 기능개발

Json 파싱을 위해서 JsonUtil 클래스 생성, toJson, fromJson 메서드 개발
toJson 메서드: Object를 Json String으로 변환
fromJson(String) 메서드: Json String을 Object로 변환
fromJson(Reader) 메서드: Reader를 받아서 라인별로 문자열을 읽어 Object로 반환
Gson을 이용하여 직렬화, 역직렬화 시행시 String 형식의 날짜 데이터를 LocalDate형식으로 변환하기 위한 inner 클래스(LocalDateTimeAdapter, LocalDateAdapter, LocalTimeAdapter) 생성

jojojojocho committed:
test: Json parsing 관련 유틸리티 클래스 테스트 코드 작성

toJson 메서드와 fromJson 메서드를 중점적으로 하여 테스트 코드 작성
이외 테스트에 필요한 inner 클래스 선언하여 사용

YurimYang commented on src/main/java/global/util/JsonUtil.java:
```diff
@@ -1,5 +1,143 @@
+    /**
+     * LocalDateTime Type의 String Format
+     */
+    private static String PATTERN_DATETIME = String.format("%s %s", PATTERN_DATE, PATTERN_TIME);
```
"departure_time": "2023-07-15T08:00:00"
LocalDateTime 형식이 이렇게 되어있어서 적용시켰을 때 "Failed making field 'java.time.LocalDateTime#date' accessible; either increase its visibility or write a custom TypeAdapter for its declaring type." 이러한 에러가 나옵니다. 혹시 이부분 수정 가능할까요 ?

jojojojocho replied:
> "departure_time": "2023-07-15T08:00:00" LocalDateTime 형식이 이렇게 되어있어서 적용시켰을 때 "Failed making field 'java.time.LocalDateTime#date' accessible; either increase its visibility or write a custom TypeAdapter for its declaring type." 이러한 에러가 나옵니다. 혹시 이부분 수정 가능할까요 ?

해당부분 수정완료했습니다!

YurimYang replied:
넵 빠른 수정 감사합니다 :) 수고하셨어요 !!

jojojojocho committed:
refactor: Json parsing 관련 유틸리티 클래스 로직 수정 및 테스트 코드 작성

LocalDateTime Stringformat이 기존 "yyyy-MM-dd HH:mm:ss"였던 것을 "yyyy-MM-dd'T'HH:mm:ss"로 변경 및 해당 로직 테스트 코드 작성

JeongUijeong commented on src/test/java/global/util/JsonUtilTest.java:
```diff
@@ -0,0 +1,231 @@
+     * TC1: 직렬화 했을 경우 문자열의 길이가 0보다 커야 한다.
+     * public static String toJson(Object obj): Object -> String
+     */
+    @Test
```
@DisplayName()과 inner 클래스를 활용하시면 테스트 결과를 확인하기 편할 것 같아요~ 이 부분 내일 회의 시간에도 다같이 테스트 코드 작성 법에 대해 간단히 다루면 좋겠네요 😀

jojojojocho replied:
> @DisplayName()과 inner 클래스를 활용하시면 테스트 결과를 확인하기 편할 것 같아요~ 이 부분 내일 회의 시간에도 다같이 테스트 코드 작성 법에 대해 간단히 다루면 좋겠네요 😀

넵 알겠습니다! 감사합니다!

JeongUijeong reviewed (approved):
수고 많으셨습니다! 직렬화, 테스트 코드 다 잘 작성해주셨네요! 👍🏻 TestCode 관련해서는 내일 회의 후 수정하셔도 될 듯 하니, 바로 approve 해드리겠습니다 🙂 코멘트에 아무 이모티콘 달아주시면 확인하신 걸로 알고 resolve 해드릴게요!
